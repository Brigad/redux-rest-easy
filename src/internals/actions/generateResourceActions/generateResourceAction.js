const generateResourceAction = (actionType, resourceName) => resourceId => ({
  type: actionType,
  resourceName,
  resourceId,
});

export default generateResourceAction;
