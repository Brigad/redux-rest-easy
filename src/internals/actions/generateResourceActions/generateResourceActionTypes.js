import generateActionType from '../../utils/generateActionType';

const generateResourceActionTypes = resourceName => ({
  INVALIDATE_RESOURCE: generateActionType(
    resourceName,
    '@invalidate',
    'INVALIDATE_RESOURCE',
  ),
  INVALIDATE_ID: generateActionType(
    resourceName,
    '@invalidate',
    'INVALIDATE_ID',
  ),
  RESET_RESOURCE: generateActionType(resourceName, '@reset', 'RESET_RESOURCE'),
});

export default generateResourceActionTypes;
