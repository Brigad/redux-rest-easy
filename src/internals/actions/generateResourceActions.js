import generateResourceAction from './generateResourceActions/generateResourceAction';
import generateResourceActionTypes from './generateResourceActions/generateResourceActionTypes';

const generateResourceActions = (resourceName) => {
  const {
    INVALIDATE_RESOURCE,
    INVALIDATE_ID,
    RESET_RESOURCE,
  } = generateResourceActionTypes(resourceName);

  return {
    resource: {
      invalidate: generateResourceAction(INVALIDATE_RESOURCE, resourceName),
      invalidateId: generateResourceAction(INVALIDATE_ID, resourceName),
      reset: generateResourceAction(RESET_RESOURCE, resourceName),
    },
  };
};

export default generateResourceActions;
