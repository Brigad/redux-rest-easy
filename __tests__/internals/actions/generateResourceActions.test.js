import generateResourceActions from '../../../src/internals/actions/generateResourceActions';

describe('generateResourceActions', () => {
  test('only path', () => {
    expect(generateResourceActions()).toMatchSnapshot();
  });
});
