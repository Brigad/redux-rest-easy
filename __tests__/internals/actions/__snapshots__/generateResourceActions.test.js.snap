// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateResourceActions only path 1`] = `
Object {
  "resource": Object {
    "invalidate": [Function],
    "invalidateId": [Function],
    "reset": [Function],
  },
}
`;
